pdr-api/handlers/parks/_identifier/name/PUT/index.js

```diff
@@ -0,0 +1,239 @@
+  changelogRecord['newStatus'] = { 'S': body.status };
+  changelogRecord['status'] = { 'S': 'historical' };
+
+  return {
+    Put: {
+      TableName: TABLE_NAME,
+      Item: changelogRecord
+    }
+  }
+}
+
+/**
+ * Updates a record in DynamoDB.
+ *
+ * @param {string} user - The user making the update.
+ * @param {Object} body - The request body.
+ * @param {string} currentTimeISO - The current time in ISO format.
+ * @returns {Promise<Object>} - A Promise that resolves to the updated attributes.
+ */
+async function updateRecord(user, body, currentTimeISO, putTransaction = undefined) {
+  // Defines the parameters for updating the record.
+  let updateParams = {
+    TableName: TABLE_NAME,
+    Key: {
+      pk: { S: body.orcs },
+      sk: { S: 'Details' }
+    },
+    ExpressionAttributeValues: {
+      ':updateDate': { S: currentTimeISO },
+      ':effectiveDate': { S: body.effectiveDate },
+      ':legalName': { S: body.legalName },
+      ':displayName': { S: body.displayName },
+      ':phoneticName': { S: body.phoneticName },
+      ':searchTerms': { S: body.searchTerms },
+      ':lastModifiedBy': { S: user },
+      ':status': { S: body.status },
+      ':notes': { S: body.notes }
+    },
+    ExpressionAttributeNames: { '#status': 'status' },
+    UpdateExpression:
+      'SET updateDate = :updateDate, #status = :status , lastModifiedBy = :lastModifiedBy, effectiveDate = :effectiveDate, legalName = :legalName, displayName = :displayName, phoneticName = :phoneticName, searchTerms = :searchTerms, notes = :notes',
+    ReturnValues: 'ALL_NEW'
+  };
+
+  try {
+    // Logs the update parameters for debugging purposes.
+    logger.debug(updateParams);
+
+    if (putTransaction) {
+      // We can't have this in a transaction.
+      delete updateParams.ReturnValues;
+      await dynamodb.transactWriteItems({
+        TransactItems: [
+          putTransaction,
+          {
+            Update: updateParams
+          }
+        ]
+      }).promise();
+
+      // Return the object we would have inserted.
+      return {
+        pk: body.orcs,
+        sk: 'Details',
+        updateDate: currentTimeISO,
+        effectiveDate: body.effectiveDate,
+        legalName: body.legalName,
+        displayName: body.displayName,
+        phoneticName: body.phoneticName,
+        searchTerms: body.searchTerms,
+        lastModifiedBy: user,
+        status: body.status,
+        notes: body.note
+      }
+    } else {
+      // Executes the update operation and retrieves the updated attributes.
+      const { Attributes } = await dynamodb.updateItem(updateParams).promise();
+
+      // Converts the DynamoDB attributes to a more usable format.
+      return AWS.DynamoDB.Converter.unmarshall(Attributes);
+    }
+  } catch (error) {
+    // Logs any errors that occur during the update operation.
+    logger.error(error);
+
+    // Propagates the error to the calling function.
+    throw error;
+  }
```

Comment:
Nit: pull the execution of the transaction outside of the function Update Record and either put it in its own function or put it inside majorChange(). Makes everything more reusable.

Reply:
I think for this maybe we can leave it for refactoring later if we have another transaction item to deal with.